private/exponential_backoff.hpp

```diff
@@ -11,51 +11,90 @@
 #include <cstdint>
 #include <random>
 
+#include "astarte_device_sdk/exceptions.hpp"
+
 namespace AstarteDeviceSdk {
 
 class ExponentialBackoff {
  public:
   /**
    * @brief Construct an ExponentialBackoff instance.
+   *
+   * @details The exponential backoff will will compute an exponential delay using
+   * the 2 as the base for the power operation.
+   * As such the values returned by calls to getNextDelay will follow the formula:
+   * min( @p initial_delay * 2 ^ ( number of calls ) , @p max_delay ) + random jitter
+   * The random jitter will be in the range [ - @p initial_delay , + @p initial_delay )
+   *
+   * @note Since the exponential backoff will be computed using @p initial_delay as a base,
+   * if that parameter is zero, then the next delay will always be zero.
+   * @note The jitter will be applied also once the @p max_delay has been reached. Effectively the
+   * maximum delay produced will be @p max_delay + @p initial_delay.
+   *
    * @param initial_delay The value for the first backoff delay.
    * @param max_delay The upper bound for all the backoff delays.
    */
   ExponentialBackoff(std::chrono::milliseconds initial_delay, std::chrono::milliseconds max_delay)
-      : initial_delay_(initial_delay), max_delay_(max_delay) {}
+      : initial_delay_(initial_delay), max_delay_(max_delay) {
+    if ((initial_delay < std::chrono::milliseconds::zero()) ||
+        (max_delay < std::chrono::milliseconds::zero())) {
+      throw AstarteInvalidInputException(
+          "Negative duration passed to the ExponentialBackoff constructor");
+    }
+    if (max_delay < initial_delay) {
+      throw AstarteInvalidInputException(
+          "Maximum delay should be larger or equal to the initial delay");
+    }
+  }
 
   /**
    * @brief Calculate and returns the next backoff delay.
-   * @details Computes the appropriate delay for the current backoff generation and increments the
-   * internal generation counter for the next call.
+   * @details See the documentation of the constructor of this class for an explanation on how
+   * this delay is computed.
    * @return The calculated delay duration.
    */
   auto getNextDelay() -> std::chrono::milliseconds {
-    constexpr double BACKOFF_FACTOR = 2.0;
+    if (initial_delay_ == std::chrono::milliseconds::zero()) {
+      return initial_delay_;
+    }
```

Comment:
Why should we return immediately if the initial_delay is equal to zero? Maybe i want to set a 0 initial_delay but with a different max_delay.